projects/packages/videopress/src/client/state/reducers.js

```diff
@@ -620,6 +621,23 @@ const playbackTokens = ( state, action ) => {
 			};
 		}
 
+		case EXPIRE_PLAYBACK_TOKEN: {
+			const { guid } = action;
+			const items = [ ...( state.items ?? [] ) ];
+			const playbackTokenIndex = items.findIndex( item => item.guid === guid );
+
+			if ( playbackTokenIndex > -1 ) {
+				// Remove it from the array
+				items.splice( playbackTokenIndex, 1 );
+			}
+
+			return {
+				...state,
+				items,
+				isFetching: false,
```

Comment:
I think we need a separate fetching flag here. `isFetching` should be set only when loading videos (and maybe renamed to something like `isFetchingVideos` to avoid the confusion).
Then, to respond to any loading state, we can define a new selector that returns `isFetchingVideos || isFetchingPlaybackToken`. Just an idea.

Reply:
Even more, I think it should be defined at the video level since the token is tied to it and not to the video list. Something similar to when a video is being deleted, for instance.

Reply:
mmm maybe it's too much. ✅

Reply:
Yeah, in principle I agree!

Reply:
> isFetching should be set only when loading videos

In this code, we are actually setting the token store `isFetching`, not the videos one. Like you said, this could have a better name, indeed :D

Reply:
> I think it should be defined at the video level since the token is tied to it and not to the video list.

The suggestion is to have an `isFetchingPlaybackToken` property on the video object? Seems interesting, may simplify things when using the flag on the actual components. Do you think the token itself could also be something that belongs to the video?

Reply:
Created #27158 to go back to this in the future.